### PR TITLE
Forbid no/abundant whitespace before array initializer

### PR DESCRIPTION
**Pull request by findepi**
Forbid these

    new int[]{...

    new int[]    {....

Extracted from https://github.com/airlift/airbase/pull/389

### CODE REVIEW

File: airbase-policy/src/main/resources/checkstyle/airbase-checks.xml [hunk 1]
```diff
@@ -256,6 +256,14 @@
             <property name="ignoreComments" value="true" />
         </module>
 
+        <module name="RegexpSinglelineJava">
+            <!-- Forbid "new Type[]{",  IntelliJ would reformat adding a space -->
+            <!-- Forbid "new Type[]  {",  IntelliJ would reformat removing a space -->
+            <property name="format" value="new\s++\w++(&lt;\?&gt;)?(\[\])++(?! )\s*+\{" />
```
Comment (martint) [inline airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:262]: Not sure I understand the last part: `(?! )\s*+\{`. How does that handle the following case: `new X[]    {`?

Comment (findepi) [reply airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:262]: it does not, that's a mistake.

Comment (findepi) [reply airbase-policy/src/main/resources/checkstyle/airbase-checks.xml]: thanks, fixed!

**Comment by findepi** [comment]
@martint do you want this rebased?

**Comment by martint** [comment]
Yes, sorry, lost track of this one. I'll take a look soon.

**Review by martint — APPROVED** [review]

**Pull request merged**